### PR TITLE
Add route to unsubscribe from all newsletters

### PR DESCRIPTION
**Pull request by phamann**
Adds a `POST /alerts/:userId/unsubscribe` api route to unsubscribe a user from all of their newsletter subscriptions.

UI to follow.

### CODE REVIEW

File: server/controllers/alerts/deleteAll.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+import logger from '@financial-times/n-logger';
+import directly from 'directly';
+import { getUserNewsletters, unsubscribeUserFromNewsletter } from '../../lib/email-web-services-client'
+
+module.exports = (req, res, next) => {
+	const userId = res.locals.userId;
+
+	logger.info(`Unsubscribing ${userId} from all alerts`, { event: 'UNSUSCRIBE_ALL', user: userId });
```
Comment (keirog) [inline server/controllers/alerts/deleteAll.js]: In case it saves headaches letter, s/SUSCRIBE/SUBSCRIBE/g

**Comment by laurieboyes** [comment]
Boom 👍

**Comment by keirog** [comment]
👍

File: server/controllers/alerts/deleteAll.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	logger.info(`Unsubscribing ${userId} from all alerts`, { event: 'UNSUBSCRIBE_ALL', user: userId });
+
+	// HACK: PH having to perform calls sequentially with directly
+	// due to race-condition with email service
+	getUserNewsletters(userId)
+		.then(lists => directly(1,
+			lists.map(list => () =>
+				unsubscribeUserFromNewsletter(userId, list.id)
+					.then((res) => {
+						logger.info(`Unsubscribed ${userId} from ${list.id}`, { event: 'UNSUSCRIBED_LIST', user: userId, list: list.id });
```
Comment (keirog) [inline server/controllers/alerts/deleteAll.js]: Same 😬

**Pull request merged**